### PR TITLE
refactor: add gas util

### PR DESCRIPTION
**Pull request by janndriessen**
## **Summary of Changes**

* Adds a gas limit utility function to simplify/centralize handling gas values over different quote functions

###### _By submitting this pull request, you are confirming the following to be true:_

- I have reviewed the [Contribution Guidelines](https://github.com/IndexCoop/index-app/blob/master/CONTRIBUTING.md).
- I have performed a self-review of my own code.
- I have updated my repository to match the master at `IndexCoop/index-app`.
- I have included test data or screenshots that prove my fix is effective or that my feature works.

### CODE REVIEW

File: src/lib/hooks/use-best-quote/index.ts [hunk 1]
```diff
@@ -175,18 +168,15 @@ export const useBestQuote = (
         ) {
           console.log('canRedeemIndexToken')
```
Comment (tokdaniel) [inline src/lib/hooks/use-best-quote/index.ts:169]: just a leftover from old times?

Comment (janndriessen) [reply src/lib/hooks/use-best-quote/index.ts:169]: yes sir - but good point will add a logs clean up PR some time in the next days.

File: src/lib/hooks/use-best-quote/utils/index-quote.ts [hunk 2]
```diff
@@ -81,24 +76,11 @@ export async function getIndexQuote(
...
-      // We don't want this function to fail for estimates here.
-      // A default will be returned if the tx would fail.
-      const canFail = false
-      const gasEstimate = await gasEstimatooor.estimate(tx, canFail)
-      console.log(gasEstimate.toString(), 'gasLimit')
-      gasLimit = gasEstimate
-    }
+    gasLimit = BigInt(estimate?.gasCosts[0].limit ?? '0')
+    gasPrice = BigInt(estimate?.gasCosts[0].price ?? '0')
     gasCosts = gasPrice * gasLimit
```
Comment (0xonramp) [inline src/lib/hooks/use-best-quote/utils/index-quote.ts:84]: interesting - it seems like this works with the string values, but should they be numbers?

**Review by 0xonramp — APPROVED** [review]
🚀

**Pull request merged**